pkg/instance/service.go

```diff
@@ -168,23 +171,26 @@ func (s service) Logs(instance *model.Instance, group *models.Group) (io.ReadClo
 	return read, err
 }
 
-func (s service) getPod(client *kubernetes.Clientset, instance *model.Instance) v1.Pod {
+func (s service) getPod(client *kubernetes.Clientset, instance *model.Instance, selector string) (v1.Pod, error) {
+	labelSelector := fmt.Sprintf("im-id=%d", instance.ID)
```

Comment:
I do really like Go's guard clause/reduce nesting style. In this case this should go into an else statement, as its a waste to compute a labelSelector if there is a non empty `selector`.